### PR TITLE
Create IndexLayerClient

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
* Create IndexLayerClient
* Add unit test

Resolves: OLPEDGE-1335
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/test/unit/IndexLayerClient.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/test/unit/IndexLayerClient.test.ts]: 2020

File: @here/olp-sdk-dataservice-read/lib/client/index.ts [hunk 2]
```diff
@@ -37,3 +37,4 @@ export * from "./QuadKeyPartitionsRequest";
 export * from "./SchemaRequest";
 export * from "./CatalogVersionRequest";
 export * from "./LayerVersionsRequest";
+export * from "./IndexLayerClient";
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/index.ts]: Could you please change header to Copyright (C) 2019 - 2020 HERE Europe B.V.

File: @here/olp-sdk-dataservice-read/test/unit/IndexLayerClient.test.ts (hunk 1 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/test/unit/IndexLayerClient.test.ts]: fixed

File: @here/olp-sdk-dataservice-read/lib/client/index.ts (hunk 2 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/client/index.ts]: Done

**Pull request merged**